### PR TITLE
Add golden and matrix tests to component template

### PR DESCRIPTION
**Pull request by bastjan**
See https://github.com/projectsyn/modulesync-control/pull/22 as to why golden tests can be a good idea.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog

### CODE REVIEW

File: commodore/component-template/{{ cookiecutter.slug }}/.gitignore [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+# Commodore
 .cache/
 helmcharts/
 manifests/
 vendor/
 jsonnetfile.lock.json
+crds/
+compiled/
+
+# Editors
```
Comment (corvus-ch) [inline commodore/component-template/{{ cookiecutter.slug }}/.gitignore:10]: This is subject to grow out of hand. We do not have an agreement on which editor to use. In consequence, every editor can be used. Adding ignores for every editor is nothing I would be willing to do.

As a git user, I consider it to be best practice, to have a global git ignore to exclude things specific to my system. This include also such things like vim swap files, the all famous and anoying `.DS_Store` macOS etc.

File: commodore/component-template/{{ cookiecutter.slug }}/Makefile [hunk 2]
```diff
@@ -49,6 +49,18 @@ docs-serve: ## Preview the documentation
 test: commodore_args += -f tests/$(instance).yml
 test: .compile ## Compile the component
 
+.PHONY: gen-golden
```
Comment (corvus-ch) [inline commodore/component-template/{{ cookiecutter.slug }}/Makefile:52]: Do we want to have the targets even if it is not enabled for the component? Consider to wrap them in conditions.

**Review by corvus-ch — CHANGES_REQUESTED** [review]
Please do not open up the think hole of adding editor specific git ignores.

File: commodore/component-template/{{ cookiecutter.slug }}/.gitignore (hunk 1 above)
Comment (simu) [reply commodore/component-template/{{ cookiecutter.slug }}/.gitignore:10]: I agree that editor-specific things don't belong in repo .gitignores but rather in a global .gitignore or (`.git/info/exclude`). However, the contents of this gitignore are verbatim from https://github.com/projectsyn/modulesync-control/blob/master/moduleroot/.gitignore.erb, so if we want to change it here, we need to change it there too.

File: commodore/component-template/{{ cookiecutter.slug }}/Makefile (hunk 2 above)
Comment (simu) [reply commodore/component-template/{{ cookiecutter.slug }}/Makefile:52]: I totally missed those when implementing the flags

File: commodore/component-template/{{ cookiecutter.slug }}/.gitignore (hunk 1 above)
Comment (simu) [reply commodore/component-template/{{ cookiecutter.slug }}/.gitignore]: Updated in latest version and removed from modulesync-control: https://github.com/projectsyn/modulesync-control/commit/668a2b3edfef9e129100a8aecda1dd08f9a891bf

File: commodore/component-template/{{ cookiecutter.slug }}/Makefile (hunk 2 above)
Comment (simu) [reply commodore/component-template/{{ cookiecutter.slug }}/Makefile:53]: added in latest version

**Review by corvus-ch — APPROVED** [review]

**Pull request merged**